src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -984,7 +984,59 @@ private BoundExpression VisitRangePatternIndexerAccess(BoundImplicitIndexerAcces
             AddPlaceholderReplacement(node.ArgumentPlaceholders[1], rangeSizeExpr);
 
             var sliceCall = (BoundCall)node.IndexerOrSliceAccess;
-            var rewrittenIndexerAccess = VisitExpression(sliceCall);
+
+            // [start..] can be simplified to .Substring(start) or .Slice(start) for some built-in types
+            // e.g. string and (ReadOnly)Span
+
+            BoundExpression? moreEfficientIndexerAccess = null;
+            if (endMakeOffsetInput is null)
+            {
+                MethodSymbol? singleArgumentOverload = null;
+                NamedTypeSymbol? typeContainingElementType = null;
+
+                if (
+                    TryGetSpecialTypeMethod(node.Syntax, SpecialMember.System_String__Substring, out var stringSubstring, isOptional: true)
+                    && sliceCall.Method.Equals(stringSubstring)
+                )
```

Comment:
Some formatting to clean up in this. For example:
```suggestion
                if (TryGetSpecialTypeMethod(node.Syntax, SpecialMember.System_String__Substring, out MethodSymbol stringSubstring, isOptional: true)
                    && sliceCall.Method.Equals(stringSubstring))
```

Reply:
Adopted your team's style (and added `Co-authored-by`)